tests/Microsoft.Azure.CosmosRepositoryAcceptanceTests/DisabledStrictTypeCheckingTests.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) IEvangelist. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.CosmosRepository;
+using Microsoft.Azure.CosmosRepository.Options;
+using Microsoft.Azure.CosmosRepositoryAcceptanceTests.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.CosmosRepositoryAcceptanceTests;
+
+/// <summary>
+/// Required as item options are cached in a static collection, the key is the Type
+/// </summary>
+public class DisabledOffer : Offer
+{
+    public DisabledOffer(string offerType, string createdBy) : base(offerType, createdBy)
+    {
+    }
+}
+
+[Trait("Category", "Acceptance")]
+[Trait("Type", "Container")]
+public class DisabledStrictTypeCheckingTests : CosmosRepositoryAcceptanceTest
+{
+    private const string OffersDatabaseName = "offers";
+    private const string OffersContainerName = "offers";
+    private const string OffersPartitionKeyPath = "/partitionKey";
+
+    private static readonly Action<RepositoryOptions> DisabledStrictTypeCheckingOptions = options =>
+    {
+        options.DatabaseId = OffersDatabaseName;
+        options.CosmosConnectionString = GetCosmosConnectionString();
+        options.ContainerPerItemType = true;
+
+        options.ContainerBuilder.Configure<DisabledOffer>(builder =>
+        {
+            builder.WithContainer(OffersContainerName);
+            builder.WithPartitionKey(OffersPartitionKeyPath);
+            builder.WithContainerDefaultTimeToLive(TimeSpan.FromMinutes(10));
+            builder.WithoutStrictTypeChecking();
```

Comment:
@Krumelur this is the proposal, you can turn this off to not force the type to be checked. However, this is at your own risk bearing in mind that the sub type MUST deserialize into the base type.

@IEvangelist what do you think? I have had this use case as well in the past and think for base types such as this. The ability to disable this feature albeit at your own risk could be very useful.

Reply:
I'm good with that, thanks for handling this.